// react-chatbot/Timestamp/Timestamp.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _DateTime = require('../utils/DateTime');

var _Timestamp = require('./Timestamp.css');

var _Timestamp2 = _interopRequireDefault(_Timestamp);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/** Displays current time according to the given format & locale */
var Timestamp = function (_PureComponent) {
  _inherits(Timestamp, _PureComponent);

  function Timestamp() {
    _classCallCheck(this, Timestamp);

    return _possibleConstructorReturn(this, (Timestamp.__proto__ || Object.getPrototypeOf(Timestamp)).apply(this, arguments));
  }

  _createClass(Timestamp, [{
    key: 'render',
    value: function render() {
      var _props = this.props,
          locale = _props.locale,
          formatOptions = _props.formatOptions;

      return _react2.default.createElement(
        'div',
        { className: _Timestamp2.default.timestamp },
        (0, _DateTime.getFormatedTime)(new Date(), locale, formatOptions)
      );
    }
  }]);

  return Timestamp;
}(_react.PureComponent);

Timestamp.displayName = 'Timestamp';

Timestamp.propTypes = {
  /** Name of the locale */
  locale: _propTypes2.default.string,
  /** Formatting options */
  formatOptions: _propTypes2.default.object
};

Timestamp.defaultProps = {
  locale: 'en-US',
  formatOptions: {
    hour: '2-digit',
    minute: '2-digit',
    hour12: true
  }
};

exports.default = Timestamp;
